packages/gatsby-theme/src/components/Share/Share.js

```diff
@@ -13,13 +14,39 @@ const Share = ({ children, ...props }) => {
       }}
       {...props}
     >
-      {children}
+      <Heading
+        sx={{
+          color: "secondary",
+          m: 0,
+          mr: "small",
+          ...sx,
+        }}
+      >
+        {heading}
+      </Heading>
+      <Box
+        __css={{
+          display: "flex",
+          alignItems: "center",
+          color: "darkShade",
+        }}
+        {...props}
+      >
+        {children}
+      </Box>
     </Box>
   )
 }
 
 Share.propTypes = {
   children: PropTypes.node.isRequired,
+  heading: PropTypes.string,
+  sx: PropTypes.shape({}),
+}
+
+Share.defaultProps = {
+  heading: undefined,
```

Comment:
Since this leaves the possibility of using the component without a Heading, we could maybe add a story without it.